Tasks/AzurePowerShellV4/Utility.ps1

```diff
@@ -1,19 +1,61 @@
+function Get-SavedModuleContainerPath {
+    return $env:SystemDrive + "\Modules";
+}
+
+function Get-SavedModuleContainerPathLinux {
+    return "/usr/share";
+}
+
+function Test-IsHostedAgentPathPresent {
+    $containerPath = Get-SavedModuleContainerPath;
+    return Test-Path "$containerPath\az_*"
+}
+
+function Test-IsHostedAgentPathPresentLinux {
+    $containerPath = Get-SavedModuleContainerPathLinux;
+    return Test-Path "$containerPath/az_*"
+}
+
 function Get-SavedModulePath {
     [CmdletBinding()]
     param([string] $azurePowerShellVersion)
-    $savedModulePath = $($env:SystemDrive + "\Modules\az_" + $azurePowerShellVersion)
+    $savedModulePath = (Get-SavedModuleContainerPath) + "\az_" + $azurePowerShellVersion;
     Write-Verbose "The value of the module path is: $savedModulePath"
     return $savedModulePath 
 }
 
 function Get-SavedModulePathLinux {
     [CmdletBinding()]
     param([string] $azurePowerShellVersion)
-    $savedModulePath =  $("/usr/share/az_" + $azurePowerShellVersion)
+    $savedModulePath =  (Get-SavedModuleContainerPathLinux) + "/az_" + $azurePowerShellVersion;
     Write-Verbose "The value of the module path is: $savedModulePath"
     return $savedModulePath
 }
 
+function Expand-ModuleZip {
+    param (
+        [string] [Parameter(Mandatory = $true)]
+        $zipPath,
+
+        [string] [Parameter(Mandatory = $true)]
+        $destination,
+
+        [bool] [Parameter(Mandatory=$true)]
+        $isWin
+    )
+    
+    if ($isWin) {
+        $parameter = @("x", "-o$destination", "$zipPath")
```

Comment:
shouldn't there be a whitespace here between -o and $destination ?

Reply:
No.
This is correct as thats the way 7zip command is defined.